projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php

```diff
@@ -62,6 +72,53 @@ public function get_data() {
 			'checklist_statuses' => get_option( 'launchpad_checklist_tasks_statuses', array() ),
 		);
 	}
+
+	/**
+	 * Updates Launchpad-related options and returns the result
+	 *
+	 * @param array $request The unsanitized request values.
+	 *
+	 * @return array Associative array with updated site options
+	 */
+	public function update_site_options( $request ) {
+		$updated = array();
+		$input   = $request->get_json_params();
+
+		foreach ( $input as $key => $value ) {
+			if ( ! is_array( $value ) ) {
+				$value = trim( $value );
+			}
+
+			switch ( $key ) {
+				case 'checklist_statuses':
+					$launchpad_checklist_tasks_statuses_option = get_option( 'launchpad_checklist_tasks_statuses' );
+
+					$filtered_input_array = array_filter(
+						(array) $value,
+						function ( $array_value ) {
+							return is_bool( $array_value );
+						}
```

Comment:
Whenever I see custom input validation/sanitization I wonder if this is something that could be handled by the API itself and triggered/documented when registering the route?

See [this endpoint](https://github.com/Automattic/jetpack/blob/e8a5954ba12f14fccb3c2f22082c966eda64d192/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php#L107-L123) as an example or [this one from Core](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php#L51-L65).